File: Code 3 Callouts/Code 3 Callouts/Models/Peds/Cop.cs
using Rage;
using LSPD_First_Response.Mod.API;
using static Stealth.Plugins.Code3Callouts.Common;

namespace Stealth.Plugins.Code3Callouts.Models.Peds
{

	internal class Cop : PedBase
	{

		internal static Cop Create(string pName, Rage.Vector3 position, float heading, bool pIsMale)
		{
			CopType pType = CopType.PoliceOfficer;

			if (Common.IsPlayerinLosSantos() == false) {
				pType = CopType.Sheriff;
			}

			if (pType == CopType.PoliceOfficer) {
				return CreateCityCop(pName, position, heading, pIsMale);
			} else {
				return CreateSheriff(pName, position, heading, pIsMale);
			}
		}

		internal static Cop Create(string pName, Rage.Vector3 position, float heading, bool pIsMale, CopType pType)
		{
			if (pType == CopType.PoliceOfficer) {
				return CreateCityCop(pName, position, heading, pIsMale);
			} else {
				return CreateSheriff(pName, position, heading, pIsMale);
			}
		}

		private static Cop CreateCityCop(string pName, Rage.Vector3 position, float heading, bool pIsMale)
		{
			if (pIsMale == true) {
				return new Cop(pName, "S_M_Y_COP_01", position, heading);
			} else {
				return new Cop(pName, "S_F_Y_COP_01", position, heading);
			}
		}

		private static Cop CreateSheriff(string pName, Rage.Vector3 position, float heading, bool pIsMale)
		{
			if (pIsMale == true) {
				return new Cop(pName, "S_M_Y_SHERIFF_01", position, heading);
			} else {
				return new Cop(pName, "S_F_Y_SHERIFF_01", position, heading);
			}
		}

		//internal Sub New(ByVal pName As String, ByVal position As Rage.Vector3)
		//    MyBase.New(pName, PedType.Cop, position)
		//End Sub

		internal Cop(string pName, Rage.Model model, Rage.Vector3 position, float heading) : base(pName, PedType.Cop, model, position, heading)
        {
			initCop();
		}

		protected internal Cop(string pName, Rage.PoolHandle handle) : base(pName, PedType.Cop, handle)
        {
			initCop();
		}

		private void initCop()
		{
			this.BlockPermanentEvents = true;
			this.RelationshipGroup = new RelationshipGroup("COP");
			this.DisplayName = "Officer";
			this.MakePersistent();
			Functions.SetCopAsBusy(this, true);
		}

		public override void Dismiss()
		{
			Functions.SetCopAsBusy(this, false);
			base.Dismiss();
		}

		internal enum CopType
		{
			PoliceOfficer,
			Sheriff
		}

	}

}

File: Code 3 Callouts/Code 3 Callouts/Models/Peds/Cop.vb
Imports Rage
Imports LSPD_First_Response.Mod.API

Namespace Models.Peds

    Public Class Cop
        Inherits PedBase

        Public Shared Function Create(ByVal pName As String, ByVal position As Rage.Vector3, ByVal heading As Single, ByVal pIsMale As Boolean) As Cop
            Dim pType As CopType = CopType.PoliceOfficer

            If Common.IsPlayerInLosSantos = False Then
                pType = CopType.Sheriff
            End If

            If pType = CopType.PoliceOfficer Then
                Return CreateCityCop(pName, position, heading, pIsMale)
            Else
                Return CreateSheriff(pName, position, heading, pIsMale)
            End If
        End Function

        Public Shared Function Create(ByVal pName As String, ByVal position As Rage.Vector3, ByVal heading As Single, ByVal pIsMale As Boolean, ByVal pType As CopType) As Cop
            If pType = CopType.PoliceOfficer Then
                Return CreateCityCop(pName, position, heading, pIsMale)
            Else
                Return CreateSheriff(pName, position, heading, pIsMale)
            End If
        End Function

        Private Shared Function CreateCityCop(ByVal pName As String, ByVal position As Rage.Vector3, ByVal heading As Single, ByVal pIsMale As Boolean) As Cop
            If pIsMale = True Then
                Return New Cop(pName, "S_M_Y_COP_01", position, heading)
            Else
                Return New Cop(pName, "S_F_Y_COP_01", position, heading)
            End If
        End Function

        Private Shared Function CreateSheriff(ByVal pName As String, ByVal position As Rage.Vector3, ByVal heading As Single, ByVal pIsMale As Boolean) As Cop
            If pIsMale = True Then
                Return New Cop(pName, "S_M_Y_SHERIFF_01", position, heading)
            Else
                Return New Cop(pName, "S_F_Y_SHERIFF_01", position, heading)
            End If
        End Function

        'Public Sub New(ByVal pName As String, ByVal position As Rage.Vector3)
        '    MyBase.New(pName, PedType.Cop, position)
        'End Sub

        Public Sub New(ByVal pName As String, ByVal model As Rage.Model, ByVal position As Rage.Vector3, ByVal heading As Single)
            MyBase.New(pName, PedType.Cop, model, position, heading)
            InitCop()
        End Sub

        Protected Friend Sub New(ByVal pName As String, ByVal handle As Rage.PoolHandle)
            MyBase.New(pName, PedType.Cop, handle)
            InitCop()
        End Sub

        Private Sub InitCop()
            Me.BlockPermanentEvents = True
            Me.RelationshipGroup = New RelationshipGroup("COP")
            Me.DisplayName = "Officer"
            Me.MakePersistent()
            Functions.SetCopAsBusy(Me, True)
        End Sub

        Public Overrides Sub Dismiss()
            Functions.SetCopAsBusy(Me, False)
            MyBase.Dismiss()
        End Sub

        Enum CopType
            PoliceOfficer
            Sheriff
        End Enum

    End Class

End Namespace